code/modules/projectiles/ammo_datums.dm

```diff
@@ -43,6 +43,8 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	var/bullet_color = COLOR_WHITE
 	///If this ammo is hitscan, the icon of beam coming out from the gun
 	var/hitscan_effect_icon = "beam"
+	///If we want to pass through dense objects
+	var/xray_proj = FALSE
```

Comment:
should be a flag in flags_ammo_behavior and bad name, this should have generic name